File: OA/wells_fargo/joseph.py
def findLastPerson(n,k):
    arr=list()
    for i in range(n):
        arr.append(i)
    index=0
    # arr.pop(index)
    while len(arr)!=1:
        index=(index+k-1)%len(arr)
        arr.pop(index)
    return arr[0]
n=int(input())
k=int(input())
print(findLastPerson(n,k))

File: OA/wells_fargo/joseph.cpp
#include<bits/stdc++.h>
using namespace std;

int findLastPerson(int n,int k)
{
    vector<int>arr(n);
    for(int i=0;i<n;i++)
    {
        arr[i]=i;
    }
    // int n=arr.size();
    int index=k-1;
    arr.erase(arr.begin()+index);
    while(arr.size()!=1)
    {
           index=(index+k-1)%arr.size();
           arr.erase(arr.begin()+index);
    }
    return arr[0];
}



int main()
{
    int n,k;
    cin>>n>>k;
    cout<<findLastPerson(n,k);
    return 0;
}